cns/restserver/ipam.go

```diff
@@ -442,28 +438,25 @@ func (service *HTTPRestService) AllocateDesiredIPConfig(podInfo cns.PodInfo, des
 }
 
 func (service *HTTPRestService) AllocateAnyAvailableIPConfig(podInfo cns.PodInfo) (cns.PodIpInfo, error) {
-	var podIpInfo cns.PodIpInfo
-
 	service.Lock()
 	defer service.Unlock()
 
 	for _, ipState := range service.PodIPConfigState {
 		if ipState.State == cns.Available {
-			_, err := service.setIPConfigAsAllocated(ipState, podInfo)
-			if err != nil {
-				return podIpInfo, err
+			if err := service.setIPConfigAsAllocated(ipState, podInfo); err != nil {
+				return cns.PodIpInfo{}, err
 			}
 
-			err = service.populateIpConfigInfoUntransacted(ipState, &podIpInfo)
-			if err != nil {
-				return podIpInfo, err
+			podIPInfo := cns.PodIpInfo{}
+			if err := service.populateIpConfigInfoUntransacted(ipState, &podIPInfo); err != nil {
+				return cns.PodIpInfo{}, err
```

Comment:
Is it normal to return this instance even with error case? 
I feel like to have *cns.PodIpInfo as return value and return nil when it has error.

Reply:
it's okay to return the zero-val with an error if it doesn't make sense to be using pointers. the caller should always check for errors first.